package-system/AWSGameLiftServerSDK/build_package_image.py

```diff
@@ -0,0 +1,289 @@
+PACKAGE_LICENSE: str = "Apache-2.0"
+PACKAGE_LICENSE_FILE: str = "LICENSE_AMAZON_GAMELIFT_SDK.TXT"
+
+GAMELIFT_SERVER_SDK_RELEASE_VERSION: str = "4.0.2"
+GAMELIFT_SERVER_SDK_DOWNLOAD_URL: str = "https://gamelift-release.s3-us-west-2.amazonaws.com/GameLift_06_03_2021.zip"
+
+PACKAGE_BASE_PATH: pathlib.Path = pathlib.Path(os.path.dirname(__file__))
+PACKAGE_ROOT_PATH: pathlib.Path = PACKAGE_BASE_PATH.parent
+PACKAGE_BUILD_TYPES: List[str] = ["Debug", "Release"]
+PACKAGE_LIB_TYPES: List[str] = ["Shared", "Static"]
+PACKAGE_PLATFORM_OPTIONS: List[str] = ["windows", "linux"]
+
+# utils
+class WorkingDirectoryInfo(object):
+    def __init__(self, root_path: pathlib.Path,
+                 source_path: pathlib.Path,
+                 build_path: pathlib.Path,
+                 output_path: pathlib.Path,
+                 libs_output_path: pathlib.Path) -> None:
+        self.root_path = root_path
+        self.source_path = source_path
+        self.build_path = build_path
+        self.output_path = output_path
+        self.libs_output_path = libs_output_path
+
+def subp_args(args) -> str:
+    arg_string = " ".join([arg for arg in args])
+    print(f"Command: {arg_string}")
+    return arg_string
+
+def delete_folder(folder: pathlib.Path) -> None:
+    shutil.rmtree(folder.resolve())
+
+# steps
+def collect_package_info() -> None:
+    parser = argparse.ArgumentParser()
+    parser.add_argument(
+        '--platform-name',
+        required=True,
+        choices=PACKAGE_PLATFORM_OPTIONS
+    )
+    
+    args = parser.parse_args()
+    global PACKAGE_PLATFORM
+    PACKAGE_PLATFORM = args.platform_name
+
+def prepare_working_directory() -> WorkingDirectoryInfo:
+    # root directory
+    root_directory: pathlib.Path = PACKAGE_BASE_PATH.joinpath("temp")
+    if root_directory.is_dir():
+        delete_folder(root_directory)
+    
+    # source and build directory
+    source_directory: pathlib.Path = \
+        root_directory.joinpath(f"GameLift-SDK-Release-{GAMELIFT_SERVER_SDK_RELEASE_VERSION}/GameLift-Cpp-ServerSDK-{PACKAGE_VERSION.split('-')[0]}")
+    build_directory: pathlib.Path = root_directory.joinpath("build")
+    build_directory.mkdir(parents=True)
+    
+    # build output and libs output directory
+    build_output_directory: pathlib.Path = PACKAGE_ROOT_PATH.joinpath(f"{PACKAGE_NAME}-{PACKAGE_PLATFORM}")
+    if build_output_directory.is_dir():
+        delete_folder(build_output_directory)
+    
+    build_libs_output_directory: pathlib.Path = build_output_directory.joinpath(f"{PACKAGE_NAME}")
+    build_libs_output_directory.mkdir(parents=True)
+    
+    return WorkingDirectoryInfo(root_directory, source_directory, build_directory, build_output_directory, build_libs_output_directory)
+
+def download_gamelift_server_sdk(working_directory: WorkingDirectoryInfo) -> None:
+    # download sdk from url
+    gamelift_sdk_zip_file: str = working_directory.root_path.joinpath("gamelift_server_sdk.zip").resolve()
```

Comment:
Nit: resolve returns a path not a string. Be careful when you are using implicit conversion like this, probably should be wrapped in str()

Reply:
will update